src/isolation/landlock.rs

```diff
@@ -0,0 +1,356 @@
+				// Should not be practically reachable.
+				_ => panic!("stat against fd {:?} failed", rawfd),
+			}
+		}
+	}
+}
+
+/// Gets the parent directory of a given host path.
+///
+/// If the file does not exist, we add the parent directory instead. This might have practical
+/// security implications, however, combined with the other security measures implemented into
+/// Uhyve, this should be fine.
+///
+/// We also derive the parent directory of files that _do_ exist, so as to give
+/// VMs the ability to remove them. See
+/// [`apply_landlock_restrictions`](crate::isolation::landlock::UhyveLandlockWrapper::apply_landlock_restrictions).
+///
+/// * `host_path` - Path whose parent directory should be derived
+fn get_parent_directory(host_path: &OsString) -> PathBuf {
+	// TODO: Make iteration amount configurable
+	// TODO: Inform user about default in docs.
+	let iterations = 2;
+	let mut host_pathbuf: PathBuf = host_path.into();
+	for _i in 0..iterations {
+		if !host_pathbuf.exists() {
+			warn!("Mapped file {:#?} not found. Popping...", host_pathbuf);
+			host_pathbuf.pop();
+			continue;
+		}
+		debug!("Adding {:#?} to Landlock", host_pathbuf);
+		return host_pathbuf;
+	}
+	panic!(
+		"The mapped file's parent directory wasn't found within {} iteration(s).",
+		iterations
+	);
+}
+
+/// Initializes Landlock using parameters provided by [`UhyveVm::new`](crate::vm::UhyveVm::new).
+///
+/// If the file does not exist, we add the parent directory instead. This might have practical
+/// security implications, however, combined with the other security measures implemented into
+/// Uhyve, this should be fine.
+///
+/// * `sandbox_mode` - File isolation sandbox mode set by the user
+/// * `kernel_path` - Location of the unikernel image
+/// * `output` - Output of Uhyve, used for adding file outputs to Landlock (if applicable)
+/// * `host_paths` - List of host paths derived from mappings by the user
+/// * `temp_dir` - Location of the temporary directory for unmapped files
+pub(crate) fn configure_landlock_uhyvevm(
+	sandbox_mode: FileSandboxMode,
+	kernel_path: String,
+	output: &crate::params::Output,
+	host_paths: &mut [OsString],
+	temp_dir: String,
+) -> UhyveLandlockWrapper {
+	// This segment adds certain paths necessary for Uhyve to function before we
+	// enforce Landlock, such as the kernel path and a couple of paths useful for sysinfo.
+	//
+	// See: https://github.com/GuillaumeGomez/sysinfo/blob/8fd58b8/src/unix/linux/cpu.rs#L420
+	//
+	// It is not necessary to whitelist e.g. /dev/kvm, as isolation should be enforced
+	// after KVM is initialized and before the kernel is loaded.
+	let uhyve_ro_paths = [
+		kernel_path,
+		String::from("/etc/"),
+		String::from("/sys/devices/system"),
+		String::from("/proc/cpuinfo"),
+		String::from("/proc/stat"),
+	]
+	.to_vec();
+
+	let mut uhyve_rw_paths: Vec<String> = [String::from("/dev/kvm")].to_vec();
+	let mut uhyve_ro_dirs: Vec<String> = Default::default();
```

Comment:
```suggestion
	let mut uhyve_ro_dirs = Vec::<String>::new();
```
Turbofish for the win!